lib.php

```diff
@@ -24,6 +24,43 @@
 
 defined('MOODLE_INTERNAL') || die();
 
+/**
+ * Adds a manage broadcast link to the course admin menu.
+ *
+ * @param navigation_node $navigation The navigation node to extend
+ * @param stdClass $course The course to object for the tool
+ * @param context $coursecontext The context of the course
+ * @return void|null return null if we don't want to display the node.
+ */
+function tool_broadcast_extend_navigation_course($navigation, $course, $coursecontext) {
+    global $PAGE;
+
+    // Only add this settings item on non-site course pages.
+    if (!$PAGE->course || $PAGE->course->id == SITEID) {
```

Comment:
This doesn't "feel" right, but it might be.
Should it be !empty($PAGE->course) as the first conditional? in case course isn't a property of $PAGE at all? and/or could we check the context level?

Reply:
> Just one comment in this one.
> Also guessing, there will be another MR to restrict the contexts users can create broadcasts in.

Yep, that's coming.

The code is pretty much a copy + paste from `tool_recyclebin` which has been out in the wild for ages.

Reply:
> This doesn't "feel" right, but it might be.
> Should it be !empty($PAGE->course) as the first conditional? in case course isn't a property of $PAGE at all? and/or could we check the context level?

See https://github.com/moodle/moodle/blob/master/admin/tool/recyclebin/lib.php#L39.